src/corehaptics.cs

```diff
@@ -0,0 +1,398 @@
+		[return: NullAllowed]
+		ICHHapticAdvancedPatternPlayer CreateAdvancedPlayer (CHHapticPattern pattern, [NullAllowed] out NSError outError);
+
+		[Export ("registerAudioResource:options:error:")]
+		nuint RegisterAudioResource (NSUrl resourceUrl, NSDictionary options, [NullAllowed] out NSError outError);
+
+		[Export ("unregisterAudioResource:error:")]
+		bool UnregisterAudioResource (nuint resourceId, [NullAllowed] out NSError outError);
+
+		[Export ("playPatternFromURL:error:")]
+		bool PlayPattern (NSUrl fileUrl, [NullAllowed] out NSError outError);
+
+		[Export ("playPatternFromData:error:")]
+		bool PlayPattern (NSData data, [NullAllowed] out NSError outError);
+	}
+
+	[Static]
+	[Internal]
+	[Mac (10,15), iOS (13,0)]
+	partial interface CHHapticPatternDefinitionKeys {
+		[Field ("CHHapticPatternKeyVersion")]
+		NSString VersionKey { get; }
+
+		[Field ("CHHapticPatternKeyPattern")]
+		NSString PatternKey { get; }
+
+		[Field ("CHHapticPatternKeyEvent")]
+		NSString EventKey { get; }
+
+		[Field ("CHHapticPatternKeyEventType")]
+		NSString EventTypeKey { get; }
+
+		[Field ("CHHapticPatternKeyTime")]
+		NSString TimeKey { get; }
+
+		[Field ("CHHapticPatternKeyEventDuration")]
+		NSString EventDurationKey { get; }
+
+		[Field ("CHHapticPatternKeyEventWaveformPath")]
+		NSString EventWaveformPathKey { get; }
+
+		[Field ("CHHapticPatternKeyEventParameters")]
+		NSString EventParametersKey { get; }
+
+		[Field ("CHHapticPatternKeyParameter")]
+		NSString ParameterKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterID")]
+		NSString ParameterIdKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterValue")]
+		NSString ParameterValueKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterCurve")]
+		NSString WeakCHHapticParameterCurveKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterCurveControlPoints")]
+		NSString WeakCHHapticParameterCurveControlPointsKey { get; }
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[StrongDictionary ("CHHapticPatternDefinitionKeys")]
+	partial interface CHHapticPatternDefinition {
+		double Version { get; set; }
+		NSArray Pattern { get; set; }
+		NSDictionary Event { get; set; }
+		CHHapticEventType EventType { get; set; }
+		double Time { get; set; }
+		double EventDuration { get; set; }
+		NSString EventWaveformPath { get; set; }
+		NSArray EventParameters { get; set; }
+		NSDictionary Parameter { get; set; }
+		// we need to do a NSString because it can be a CHHapticEventParameterID or a CHHapticDynamicParameterID 
+		// which are two different enums. User will have to do the conversion
+		[Advice ("Value can be either a CHHapticEventParameterId or a CHHapticDynamicParameterId. A check is needed against both enums.")]
+		NSString ParameterId { get; set; }
+		double ParameterValue { get; set; }
+		// we do not have docs or header information about the exact type
+		NSObject WeakCHHapticParameterCurve { get; set; }
+		NSObject WeakCHHapticParameterCurveControlPoints { get; set; }
```

Comment:
same as above: the `CHHaptic` text should **not** be part of the public name

if we add _strong_ versions later then the names should be `WeakFoo` and `Foo`, not `WeakCHHapticFoo` and `Foo`